packages/services/api/src/modules/schema/providers/registry-checks.ts

```diff
@@ -139,6 +154,72 @@ export class RegistryChecks {
     } satisfies CheckResult;
   }
 
+  async policyCheck({
+    orchestrator,
+    project,
+    schemas,
+    selector,
+    modifiedSdl,
+  }: {
+    orchestrator: Orchestrator;
+    schemas: [SingleSchema] | PushedCompositeSchema[];
+    project: Project;
+    modifiedSdl: string;
+    selector: {
+      organization: string;
+      project: string;
+      target: string;
+    };
+  }) {
+    const sdl = await ensureSDL(
+      orchestrator.composeAndValidate(
+        schemas.map(s => this.helper.createSchemaObject(s)),
+        project.externalComposition,
+      ),
+    );
+
+    try {
+      const policyResult = await this.policy.checkPolicy(sdl.raw, modifiedSdl, selector);
+      const warnings =
+        policyResult?.warnings?.map<SchemaCheckWarning>(record => ({
+          message: record.message,
+          source: record.ruleId ? `policy-${record.ruleId}` : 'policy',
+          column: record.column,
+          line: record.line,
+        })) ?? [];
+
+      if (policyResult === null) {
+        return {
+          status: 'skipped',
+        } satisfies CheckResult;
+      }
+
+      if (policyResult.success) {
+        return {
+          status: 'completed',
+          result: {
+            warnings,
+          },
+        } satisfies CheckResult;
+      }
+
+      return {
+        status: 'failed',
+        reason: {
+          errors: policyResult.errors,
+          warnings,
+        },
+      } satisfies CheckResult;
+    } catch (e) {
+      return {
+        status: 'failed',
+        reason: {
+          errors: [{ message: (e as Error).message }],
```

Comment:
to be honest I use the same pattern `(e as Error).message` without checking of `instaneof Error` :)